### PR TITLE
Try another memcpy optimisation for small blobs

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
todo

### CODE REVIEW

File: base/glibc-compatibility/memcpy/memcpy.h [hunk 1]
```diff
@@ -214,3 +217,280 @@ static inline void * inline_memcpy(void * __restrict dst_, const void * __restri
...
+
+
+template <uint8_t SIZE>
+static inline void * MemcpyFixed(void * dest, const void * src)
+{
+    return detail::inline_memcpy(dest, src, SIZE);
+}
+
+/// The idea is taken from https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/fast_mem.hpp
+static inline void * inline_memcpy(void * __restrict dest, const void * __restrict src, size_t size)
```
Comment (alexey-milovidov) [inline base/glibc-compatibility/memcpy/memcpy.h:231]: But overlapping stores should not be worse.
The technique of overlapping stores is used by glibc, cosmopolitan libc, and folly.

Comment (nickitat) [reply base/glibc-compatibility/memcpy/memcpy.h:231]: we'll still use this technique. the optimisation only tries to save smth by generating hopefully better instantiations of function with substituted `size`

Comment (alexey-milovidov) [reply base/glibc-compatibility/memcpy/memcpy.h:231]: For small sizes, the compiler will completely replace `memcpy` with its own code:
https://godbolt.org/z/zKMThMvWb

Comment (alexey-milovidov) [reply base/glibc-compatibility/memcpy/memcpy.h:231]: About 1% slowdown: https://s3.amazonaws.com/clickhouse-test-reports/44243/f48f4be976d969c6709ea7b49f2dcb6aa351b198/performance_comparison_[2/4]/report.html

**Comment by nickitat** [comment]
experiment was unsuccessful

**Pull request closed without merging**